test/unit/test_str.c

```diff
@@ -3,6 +3,34 @@
 
 //TODO test r_str_chop_path
 
+bool test_r_str_format(void) {
+	r_strf_frame (32);
+	const char *a = r_strf ("%s.%d", "hello", 42);
+	mu_assert_streq (a, "hello.42", "error in r_strf hello.42");
+
+	const char *b = r_strf ("%s.%d", "world", 24);
+	mu_assert_streq (b, "world.24", "error in r_strf world.24");
+
+	const char *c = r_strf ("%s.%d", "finds", 23);
+	mu_assert_streq (c, "finds.23", "error in r_strf finds.23");
+
+	mu_assert_streq (a, "hello.42", "error in r_strf hello.42");
+	mu_assert_streq (b, "world.24", "error in r_strf world.24");
+
+	mu_end;
+}
+
+bool test_r_str_format_limits(void) {
+        r_strf_frame (9);
+        const char *a = r_strf ("%s.%d", "hello", 42);
```

Comment:
what if you have to use both strings at the same time? e.g. `sdb_set(sdb, r_strf(...), r_strf(...))`

You would have no way to know whether one `r_strf` overwrites the content of the other.

Reply:
Yes and once it asserts, what can you do? Asserting means you have done something wrong at *programming time*, but that's not really true. It's just that, sometimes, this implementation will fail. That's why I don't like this.